Algorithms/Strings/RabinKarp.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+
+namespace Algorithms.Strings
+{
+    /// <summary>
+    /// The idea: You calculate the hash for the pattern <c>p</c> and the hash values for all the prefixes of the text <c>t</c>.
+    /// Now, you can compare a substring in constant time using the calculated hashes.
+    /// time complexity: O(p + t),
+    /// space complexity: O(t),
+    /// where   t - text length
+    ///         p - pattern length.
+    /// </summary>
+    public static class RabinKarp
+    {
+        /// <summary>
+        /// Finds the index of all occurrences of the pattern <c>p</c> int <c>t</c>.
+        /// </summary>
+        /// <param name="t">Input text.</param>
+        /// <param name="p">Search pattern.</param>
+        /// <returns>List of starting indices of the pattern in the text.</returns>
+        public static List<int> FindAllOccurrences(string t, string p)
```

Comment:
Codacy found an issue: [Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6278031)